sqlx-core/src/sqlite/connection/mod.rs

```diff
@@ -146,6 +151,14 @@ impl Connection for SqliteConnection {
 
     fn close(mut self) -> BoxFuture<'static, Result<(), Error>> {
         Box::pin(async move {
+            if self.optimize_on_close {
+                let mut pragma_string = String::new();
+                if let Some(limit) = self.analysis_limit {
+                    pragma_string.push_str(&format!("PRAGMA analysis_limit = {}; ", limit));
```

Comment:
So this is kind of what I was talking about here: https://github.com/launchbadge/sqlx/issues/2111#issuecomment-1256729525

`analysis_limit` _can_ be set with the other settings pragmas on startup so it affects all `ANALYZE` calls. The way I wrote the doc comment for it in the original issue is meant to imply that's what it does.

If we _only_ want it to affect the `PRAGMA optimize;` call then it should be an argument to `optimize_on_close()` and not a separate setter. I was thinking we could have both.

Reply:
Thanks for the tips.